tests/keithley_2700.py

```diff
@@ -0,0 +1,168 @@
+import unittest
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import get_default_ioc_dir
+from utils.test_modes import TestModes
+from utils.testing import get_running_lewis_and_ioc, skip_if_recsim
+import time
+
+DEVICE_PREFIX = "KHLY2700_01"
+
+IOCS = [
+    {
+        "name": DEVICE_PREFIX,
+        "directory": get_default_ioc_dir("KHLY2700"),
+        "macros": {},
+        "emulator": "keithley_2700",
+    },
+]
+
+TEST_MODES = [TestModes.RECSIM, TestModes.DEVSIM]
+
+
+class Status(object):
+    ON = "ON"
+    OFF = "OFF"
+
+
+class Keithley_2700Tests(unittest.TestCase):
+    """
+    Tests for the Keithley2700.
+    """
+
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc("keithley_2700", DEVICE_PREFIX)
+        self.ca = ChannelAccess(default_timeout=30, device_prefix=DEVICE_PREFIX)
+        self.ca.wait_for("IDN")
+
+    def test_GIVEN_scan_state_set_WHEN_read_THEN_scan_state_is_as_expected(self):
```

Comment:
Two points here

- I think your `GIVEN` is a `WHEN`. The test should describe behaviour, not functions. `WHEN_I_set_the_scan_state_THEN_the_readback_matches_the_set_state`. If there's no special context for the test, then the `GIVEN` can be omitted
- One of my pet peeves in tests, ask @Tom-Willemsen, is "as expected" or "is correct". It doesn't add any worthwhile information to the test name. A test name should be self-describing in terms of expected behaviour. "as expected" simply confers that this is a test with an expected result but doesn't confer anything about the system under test.

Please apply throughout.

Reply:
Changed test titles